lib/ui/view/channel/new_channel/new_channel.dart

```diff
@@ -57,54 +70,202 @@ class _NewChannelState extends State<NewChannel> {
+                        width: 0.5,
+                      ),
+                    ),
+                    child: Row(
+                      children: [
+                        Text(
+                          '# ',
+                          style: AppTextStyles.body2Medium.copyWith(
+                            color: AppColors.darkGreyColor,
+                          ),
+                        ),
+                        Expanded(
+                          child: TextField(
+                            //controller: _channelName,
+                            style: AppTextStyles.body2Medium.copyWith(
+                              color: AppColors.blackColor,
+                            ),
+                            inputFormatters: [
+                              LengthLimitingTextInputFormatter(80)
+                            ],
+                            cursorColor: AppColors.zuriPrimaryColor,
+                            decoration: InputDecoration(
+                              border: InputBorder.none,
+                              hintText: 'e.g team-uchiha',
+                              hintStyle: TextStyle(
+                                fontWeight: FontWeight.w400,
+                                color: Color(0xFFA1A9B2),
+                              ),
+                            ),
+
+                            onChanged: (val) {
+                              setState(() {
+                                validateInput(val);
+                                _maxLength = 80 - val.length;
+                              });
+                            },
+                          ),
+                        ),
+                        Text(
+                          '${_maxLength}',
+                          style: TextStyle(
+                            fontWeight: FontWeight.w400,
+                            color: Color(0xFFA1A9B2),
+                          ),
+                        )
+                      ],
+                    ),
+                  ),
                   Padding(
-                    padding: const EdgeInsets.all(8.0),
-                    child: TextField(
-                      //controller: _channelName,
-                      decoration: InputDecoration(
-                          border: const OutlineInputBorder(),
-                          prefixIcon: RichText(
-                            text: TextSpan(
-                                text: '  #',
-                                style: AppTextStyles.body3Medium,
-                                children: <TextSpan>[
-                                  TextSpan(
-                                    text: ' e.g team Uhciha',
-                                    style: AppTextStyles.body2Medium,
-                                  ),
-                                ]),
+                    padding: const EdgeInsets.only(left: 16.0),
+                    child: Visibility(
+                      visible: _hasError,
```

Comment:
_hasError is suppose to be a viewmodel logic, so we can test all this.

move the logic there, ensure your UI have no logic